test/performance/range/gap_decorator_seq_write_benchmark.cpp

```diff
@@ -0,0 +1,119 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <chrono>
+#include <cmath>
+#include <cstring>
+#include <random>
+#include <utility>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alignment/exception.hpp>
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/range/decorator/all.hpp>
+#include <seqan3/std/ranges>
+
+#include "gap_decorator_helper.hpp"
+
+using namespace seqan3;
+
+// ============================================================================
+//  insert left to right
+// ============================================================================
+template <typename gap_decorator_t, bool gapped_flag>
+void insert_left2right(benchmark::State& state)
+{
+    unsigned int seq_len = state.range(0);
+    using size_type = typename gap_decorator_t::size_type;
+    using sequence_type = remove_cvref_t<detail::unaligned_seq_t<gap_decorator_t>>;
+    sequence_type seq(seq_len, 'A'_dna4);
+
+    // vector of sampled gap lengths for each position
+    std::vector<size_type> gaps(seq_len, 0);
+
+    // determine sum of gaps and non-gap symbols for not exceeding targeted sequence length
+    if (gapped_flag)
+    {
+        sample<size_type>(gaps, seq_len, state.range(1)/100.0);
+        resize<size_type, sequence_type>(gaps, seq, seq_len);
+    }
+    // initialize with (truncated) sequence and insert gaps from left to right
+    gap_decorator_t gap_decorator;
+    assign_unaligned(gap_decorator, seq);
+
+    // insert gaps before starting benchmark
+    if (gapped_flag)
+        insert_gaps<gap_decorator_t>(gaps, gap_decorator, seq_len);
+
+    auto it = std::ranges::begin(gap_decorator);
+    for (auto _ : state)
+    {
+        if (it == std::ranges::end(gap_decorator))
+            it = std::ranges::begin(gap_decorator);
+        it = insert_gap(gap_decorator, it, 1);
+        ++it;
```

Comment:
maybe skip one more / increment twice?

Reply:
Yes, for not having one contiguous gap.

Reply:
But this also adds another condition in the if-clause (for which the timer is not paused) in order to avoid the iterator to jump over the end.

Reply:
But this also adds another condition into the if-clause for avoiding that the iterator jumps over the end and will therefore affect the timer.

Reply:
Well, right now the check is not needed, is it? Because you would never reach the end...

Reply:
With `++++it` I will need the check, except I insert 2 gaps. Should I?

Reply:
I'll do it.